### PR TITLE
[fix](Nereids) join order is not right after sql parsing

### PR DESCRIPTION
**Pull request by morrySnow**
## Proposed changes

Issue Number: close #xxx

for sql
```
t1, t2 join t3
```

we should generate plan like:
```
t1 join (t2 join t3)
```

but we generate:
```
(t1 join t2) join t3
```
to follow legancy planner.

this PR change Nereids to correct behavior.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run pipelinex_p0

**Review by starocean999 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java [hunk 1]
```diff
@@ -2808,7 +2808,7 @@ private LogicalPlan withRelations(LogicalPlan inputPlan, List<RelationContext> r
         LogicalPlan left = inputPlan;
         for (RelationContext relation : relations) {
             // build left deep join tree
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java:2810]: 左深变右深了，注释还是左深

**Review by englefly — APPROVED** [review]

**Pull request merged**